### PR TITLE
Watch Cluster CR explicitly instead of watching owner

### PR DESCRIPTION
**Pull request by pvsune**
We are watching `Cluster` by `builder.Owns`; this assumes `Console` owns `Cluster`. But we don't add `controllerutil.SetControllerReference(&console, &cluster, scheme)` anywhere. This will also, in effect, delete the `Cluster` when `Console` is deleted.

We are also setting `spec.ClusterKeyRef` to reference the `Cluster` connected to by `Console`. If we want to enqueue requests for `Console` by watching `Cluster` without setting the owner, we will need to index `spec.ClusterKeyRef`. If `Console` to `Cluster` connection is one-to-one, it is simpler to assume the name and namespace of both `Console` and `Cluster` are the same.

If `Cluster` is not found (e.g. deleted), the reconciliation will never end. We should probably ignore this and set the message in `Console` status.

This PR introduces the following:
- Index `spec.clusterKeyRef.name` so we can filter Consoles
- Watch `Cluster` explicitly instead of assuming `Console` owns `Cluster`.
- Ignore reconcile request if `Cluster` is not found and set the condition
- Add `status.conditions` in the Console CRD

Note:
- We will need to `make manifests` to regenerate `Console` CR but there is a problem with `cloudhut/common` not having json tags. See #5

### CODE REVIEW

**Comment by RafalKorepta** [comment]
> it is simpler to assume the name and namespace of both Console and Cluster are the same.

Let's not assume that. We should give our users ability to install consoles and clusters separately. 

> If Cluster is not found (e.g. deleted), the reconciliation will never end. 

In my opinion if cluster is not found 
https://github.com/RafalKorepta/redpanda/blob/e812d191feb43a49fa090fb6aa15d9ae9d7e1cbd/src/go/k8s/controllers/redpanda/console_controller.go#L77

This will stop reconciliation loop, right?

> We should probably ignore this and set the message in Console status.

I agree that we should set the status in the Console CR that cluster is not found.

File: src/go/k8s/controllers/redpanda/console_controller.go [hunk 1]
```diff
@@ -452,7 +459,7 @@ func getDeploymentStrategy(console *redpandav1alpha1.Console) v1.DeploymentStrat
 func (r *ConsoleReconciler) SetupWithManager(mgr ctrl.Manager) error {
 	return ctrl.NewControllerManagedBy(mgr).
 		For(&redpandav1alpha1.Console{}).
-		Owns(&redpandav1alpha1.Cluster{}).
```
Comment (RafalKorepta) [inline src/go/k8s/controllers/redpanda/console_controller.go:455]: This is right change. I copied from cluster controller that will not work in this setup.

File: src/go/k8s/controllers/redpanda/console_controller.go [hunk 2]
```diff
@@ -71,9 +74,13 @@ func (r *ConsoleReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ct
 
 	var cluster redpandav1alpha1.Cluster
 	if err := r.Get(ctx, types.NamespacedName{
-		Namespace: console.Spec.ClusterKeyRef.Namespace,
-		Name:      console.Spec.ClusterKeyRef.Name,
+		Namespace: console.GetNamespace(),
+		Name:      console.GetName(),
```
Comment (RafalKorepta) [inline src/go/k8s/controllers/redpanda/console_controller.go]: We shouldn't assume that. We should be flexible.

File: src/go/k8s/controllers/redpanda/console_controller.go [hunk 3]
```diff
@@ -71,9 +74,13 @@ func (r *ConsoleReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ct
...
 	}, &cluster); err != nil {
+		if apierrors.IsNotFound(err) {
+			// WANT: We should probably update status.condition here.
+			return ctrl.Result{}, nil
+		}
```
Comment (RafalKorepta) [inline src/go/k8s/controllers/redpanda/console_controller.go]: I agree we should. You can add condition as Nicola did in cluster CR. 

REF https://github.com/redpanda-data/redpanda/pull/3978

File: src/go/k8s/apis/redpanda/v1alpha1/console_types.go [hunk 4]
```diff
@@ -31,10 +31,7 @@ type ConsoleSpec struct {
 	MessagePack msgpack.Config `json:"messagePack"`
 	Protobuf    proto.Config   `json:"protobuf"`
 
-	// ClusterKeyRef references to Cluster Custom Resource which will
-	// set kafka.Config configuration in Redpanda console
-	ClusterKeyRef corev1.ObjectReference `json:"clusterKeyRef"`
-	ClientID      string                 `json:"clientID,omitempty"`
```
Comment (RafalKorepta) [inline src/go/k8s/apis/redpanda/v1alpha1/console_types.go]: I wouldn't remove reference from console CR

**Review by RafalKorepta — CHANGES_REQUESTED** [review]

File: src/go/k8s/apis/redpanda/v1alpha1/console_types.go (hunk 4 above)
Comment (pvsune) [reply src/go/k8s/apis/redpanda/v1alpha1/console_types.go]: Updated

File: src/go/k8s/controllers/redpanda/console_controller.go (hunk 2 above)
Comment (pvsune) [reply src/go/k8s/controllers/redpanda/console_controller.go]: Updated to index fields and reconcile cluster from `clusterKeyRef`

File: src/go/k8s/controllers/redpanda/console_controller.go (hunk 3 above)
Comment (pvsune) [reply src/go/k8s/controllers/redpanda/console_controller.go]: Add conditions in `ConsoleStatus`

**Comment by pvsune** [comment]
> This will stop reconciliation loop, right?

Since we are throwing an error this will requeue again.

**Pull request closed without merging**